crates/re_space_view_text_log/src/space_view_class.rs

```diff
@@ -402,16 +406,60 @@ fn table_ui(
 
                 // body
                 row.col(|ui| {
-                    let mut text = egui::RichText::new(entry.body.as_str());
+                    if state.selectable {
+                        // Use a &str with TextEdit so body is selectable but immutable.
+                        let entry_text_rows = entry.num_body_lines;
+                        let mut binding = entry.body.as_str();
+                        let mut text = egui::TextEdit::multiline(&mut binding)
+                            .min_size([0.0, 0.0].into())
+                            .desired_rows(entry_text_rows)
+                            .desired_width(f32::INFINITY);
+                        // Font information for LayoutJob, based on default_layouter in `egui::TextEdit::show_contents`
+                        let font_id = if state.monospace {
+                            egui::TextStyle::Monospace.into()
+                        } else {
+                            egui::FontSelection::default()
+                        }
+                        .resolve(ui.style());
+                        let text_color = if let Some(color) = entry.color {
+                            color.into()
+                        } else {
+                            // TODO(lupickup): Understand why TextEdit text color is white instead of gray like the RichText version.
+                            ui.visuals()
+                                .override_text_color
+                                .unwrap_or_else(|| ui.visuals().widgets.inactive.text_color())
+                        };
+                        let mut layouter = |ui: &egui::Ui, string: &str, wrap_width: f32| {
+                            let mut layout_job = egui::text::LayoutJob::simple(
+                                string.to_owned(),
+                                font_id.clone(),
+                                text_color,
+                                wrap_width,
+                            );
+                            layout_job.wrap.max_rows = entry_text_rows;
+                            // Fill to end of body column instead of choosing a full "word" to ellide
+                            layout_job.wrap.break_anywhere = true;
+                            ui.fonts(|f| f.layout_job(layout_job))
```

Comment:
I suggest you compute this `Galley` once and copy into this `layouter`. The `string` you know (and is fixed), and `wrap_width` should be `f32::INFINITY` (no wrapping).

By computing the galley up-front you can use the size of it to size the `TextEdit`.